src/parseTools.js

```diff
@@ -894,7 +842,7 @@ function getHeapForType(type, unsigned) {
 // case external JS library code uses this name.
 function makeStructuralReturn(values) {
   assert(values.length == 2);
-  return 'setTempRet0(' + values[1] + '); return ' + asmCoercion(values[0], 'i32');
+  return 'setTempRet0(' + values[1] + '); return ' + values[0];
```

Comment:
There is a possible risk here, as this turns `return x | 0` into `return x`, and maybe `x` was unsigned.

The risk is probably low, but I'm not sure how low?

Reply:
Is this a risk just here for `makeStructuralReturn`?  Or elsewhere too?  I'm not sure I understand that the risk exactly.. maybe you could give an example of how it could break?

Internally makeStructuralReturn only has a single user `__cxa_find_matching_catch` and `values[0]` is always a pointer in that case.

Reply:
It could be in the others as well.

If we used to have
```javascript
function foo() {
  return x + y | 0;
```
but now we have
```javascript
function foo() {
  return x + y;
```
Then the result may be different, if it overflows beyond a 32-bit integer. It seems unlikely, but it's still a risk. Leaving the coercion is probably safest.

Reply:
It does sound like the single user here is fine, though. If you audit all the other code paths in the others as well, then sgtm.